flink-ml-api/src/main/java/org/apache/flink/ml/linalg/Vector.java

```diff
@@ -29,6 +29,10 @@
     /** Gets the value of the ith element. */
     double get(int i);
 
+
+    /** set the value of the ith element. */
+    void set(int i, double val);
```

Comment:
I am not sure we need a `set` method for `Vector`. From my perspective it might be better to make `Vector` immutable and recreate a new `Vector` if we need to change it.

Reply:
In some case, we just want to update the vector, we use sample to update special value. if use vector new, it may allocate a new memory, it not always the good thing.

Reply:
Hmm... I still cannot agree with this. I read the usage of `Vector.set()` in Knn and I think we can satisfy the need to update a `Vector` in knn without adding much memory burden by creating and updating a double array before using the array to create a `Vector`.

Reply:
done